src/history/history.go

```diff
@@ -168,3 +168,14 @@ func Services() map[string]State {
 	mux.Unlock()
 	return historyClone
 }
+
+// CheckSubdomain checks if the subdomain is in use
+func CheckSubdomain(subdomain string) bool {
+	currentStatus := Services()
```

Comment:
Why do you need a copy of the services? You can use the mux.Lock() and then check.
just do `defer mux.Unlock()`

Reply:
If you do the check in the `SetState()` function, make sure you unexport this function and don't call mux.Lock() as it will deadlock otherwise.